veinminer-server/src/main/java/wtf/choco/veinminer/tool/ToolCategoryRegistry.java

```diff
@@ -68,19 +68,79 @@ public VeinMinerToolCategory get(@NotNull ItemType itemType, @NotNull Predicate<
     }
 
     /**
+     *
+     * Deprecated: Using this leads to incorrect behaviour when an NBT tag is set in config.
+     * Get the {@link VeinMinerToolCategory} that contains the given {@link ItemType}.
+     * Pass ItemStack instead of ItemType. There is no guarantee as to which category will
+     * be returned if more than one category contains the provided ItemType.
+     *
+     * @param itemType the item type
+     *
+     * @return the corresponding tool category, or null if no category contains the item
+     */
+    @Deprecated
+    @Nullable
+    public VeinMinerToolCategory get(@NotNull ItemType itemType) {
+        return get(itemType, null, PREDICATE_ALWAYS_TRUE);
+    }
+
+    /**
+     *
+     * Deprecated: Using this leads to incorrect behaviour when an NBT tag is set in config.
+     * Get the {@link VeinMinerToolCategory} that contains the given {@link ItemType}.
+     * Pass ItemStack instead of ItemType. There is no guarantee as to which category will
+     * be returned if more than one category contains the provided ItemType.
+     *
+     * @param itemType the item type
+     * @param categoryPredicate a predicate to apply on top of the item condition. If the predicate
+     * returns false for any given category, it will not be returned by this method. Useful for an
+     * additional permission check on a category
+     *
+     * @return the corresponding tool category, or null if no category contains the item
+     */
+    @Deprecated
+    @Nullable
+    public VeinMinerToolCategory get(@NotNull ItemType itemType, @NotNull Predicate<VeinMinerToolCategory> categoryPredicate) {
+        return get(itemType, null, categoryPredicate);
+    }
+
+
+
```

Comment:
Some extra whitespace lines here. Ideally just one line of spacing between methods please :) Same goes for other places in this file